### PR TITLE
[query] Move trace handler to server level

### LINKED ISSUE #6141: Move traceResponseHandler
https://github.com/jaegertracing/jaeger/pull/6121/files#r1818135953

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Which problem is this PR solving?
- Resolves #6141

## Description of the changes
- Move the trace response handler to the server level so that it is applied to the whole server rather than just within the API handler. 

## How was this change tested?
- Unit test

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: cmd/query/app/trace_response_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	req, err := http.NewRequest(http.MethodGet, server.URL, nil)
+	require.NoError(t, err)
+
+	resp, err := server.Client().Do(req)
+	require.NoError(t, err)
+
+	traceResponse := resp.Header.Get("traceresponse")
+
+	// TODO: why isn't this populated?
+	require.Equal(t, "", traceResponse)
```
Comment (mahadzaryab1) [inline cmd/query/app/trace_response_handler_test.go:48]: @yurishkuro do you know what's wrong with the setup of my test? this is empty at the moment

Comment (mahadzaryab1) [reply cmd/query/app/trace_response_handler_test.go:48]: I think I need to inject the traceparent header - giving that a shot

Comment (yurishkuro) [reply cmd/query/app/trace_response_handler_test.go:48]: don't know, looks correct. Try log something for debugging

**Comment by mahadzaryab1** [comment]
unit test failure is transient i believe - rerun should fix it

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**